02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,355 @@
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(hcop)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE71270.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE71270.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene ID column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+## Mapping zebrafish Ensembl gene IDs to human symbols
+
+refine.bio data uses Ensembl gene identifiers, which will be in the first column.
+
+```{r}
+# Let's take a look at the first 6 items in the first column of `df`
+head(df[, 1])
+```
+
+In zebrafish, Ensembl gene identifiers begin with `ENSDARG`.
+
+Now let's do the conversion!
+We'll use the `zebrafish` data from `hcop`; this is what `zebrafish` looks like.
+
+```{r}
+zebrafish
+```
+
+We're interested in the `human_symbol` and `zebrafish_ensembl` columns specifically.
+
+```{r}
+human_symbol_df <- zebrafish %>%
+  # we'll want to subset zebrafish to only the columns we're interested in
+  dplyr::select(zebrafish_ensembl, human_symbol) %>%
+  # only retain the genes from the expression matrix that are in zebrafish data
+  dplyr::inner_join(df, by = c("zebrafish_ensembl" = "Gene"))
+```
+
+Here's what the new data frame looks like:
+
+```{r}
+head(human_symbol_df, 25)
+```
+
+Let's get a summary of the gene symbols returned in the `human_symbol` column of our mapped data frame.
+
+```{r}
+# We can use the `summary()` function to get a better idea of the distribution of symbols in the `Symbol` column
+summary(as.factor(human_symbol_df$human_symbol)) # We need to use the `as.factor()` function here in order to get the count of each unique value returned
+```
+
+Looks like we have mapped symbols!
+It does not explicitly tell us if we have IDs that did not map to symbols, which would result in NAs so let's explore to find out if we do have any.
+
+```{r}
+any(is.na(human_symbol_df$human_symbol)) # Logical statement to check if there are any NAs in the `human_symbol` column of our mapped data frame
+```
+
+There are no NAs!
```

Comment:
I find it unlikely (maybe impossible?) that a no NAs situation would happen when mapping across species. So that's what was suspicious to me and made me look again at your `inner_join()` step.